src/gradio/app/main.py

```diff
@@ -112,64 +126,70 @@ async def generate_post_async(
     global llm_tasks
 
     if not summary:
-        return "Summarization is still in progress. Please wait and try again."
-
-    post_text = prepare_post_text(
-        summary,
-        audience,
-        english_level,
-        length,
-        hashtag_preference,
-        perspective,
-        emoji_usage,
-        question_option,
-        paper_url,
-        custom_requirements,
-    )
+        raise HTTPException(status_code=400, detail="Summarization not completed")
+    else:
+        post_text = prepare_post_text(
+            summary,
+            audience,
+            english_level,
+            length,
+            hashtag_preference,
+            perspective,
+            emoji_usage,
+            question_option,
+            paper_url,
+            custom_requirements,
+        )
 
-    llm = Ollama(
-        model=settings.generation_ollama_model,
-        base_url=settings.ollama_url,
-    )
+        llm = Ollama(
+            model=settings.generation_ollama_model,
+            base_url=settings.ollama_url,
+        )
 
-    try:
-        loop = asyncio.get_running_loop()
-        llm_task = loop.run_in_executor(None, llm.invoke, post_text)
-        llm_tasks[session_id] = llm_task
-        generated_post = await llm_task
-        return generated_post
-    except asyncio.CancelledError:
-        logging.warning(f"LLM processing was cancelled for session {session_id}.")
-        return "LLM processing was stopped by the user."
-    finally:
-        llm_tasks.pop(session_id, None)
+        try:
+            loop = asyncio.get_running_loop()
+            llm_task = loop.run_in_executor(None, llm.invoke, post_text)
+            llm_tasks[session_id] = llm_task
+            generated_post = await llm_task
+            return generated_post
+        except Exception as e:
+            logging.error(f"Failed during processing: {str(e)}")
+            raise HTTPException(
+                status_code=500, detail=f"Failed to process PDF: {str(e)}"
+            )
+        except asyncio.CancelledError:
+            logging.warning(f"Summarization was cancelled for session {session_id}.")
+            raise HTTPException(status_code=400, detail="Summarization was cancelled")
```

Comment:
Replace raising error by either gr.Info("Text generation was cancelled") or just simply omit the exception

Reply:
`gr.info()` does not exist and cannot be used to inform the user. Do you mean print asyncio.CancelledError?

Reply:
Same as above